Add UI tests for Stellar Burgers

Gabaraeva opened the pull request:
"UI тесты для Stellar Burgers:

Gabaraeva committed:
Add UI tests for Stellar Burgers

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,37 @@
+@pytest.fixture
+def create_order(driver):
+    # Реализация создания заказа
+    def _create_order():
```
Нужно исправить: данная фикстура не выполняет никакой логики. Необходимо добавить реализацию

ShamilovR commented on pages/base_page.py:
```diff
@@ -0,0 +1,31 @@
+class BasePage:
+    def __init__(self, driver):
+        self.driver = driver
+        self.base_url = "https://stellarburgers.nomoreparties.site"
```
Нужно исправить: базовый класс это абстракция. Он не должен работать с конкретными элементами, локаторами и урлами.

ShamilovR commented on pages/main_page.py:
```diff
@@ -0,0 +1,43 @@
+    MODAL_CLOSE_BUTTON = (By.XPATH, "//div[@class='Modal_modal__P3_V5']/button")
+    CONSTRUCTOR_SECTION = (By.ID, "constructor")
+
+    def open(self):
```
Нужно исправить: здесь и далее: отсутствует декоратор step над методом. Его необходимо прописывать над методами пейджей для информативного allure отчета.

ShamilovR commented on pages/main_page.py:
```diff
@@ -0,0 +1,43 @@
+    CONSTRUCTOR_SECTION = (By.ID, "constructor")
+
+    def open(self):
+        self.driver.get(f"{self.base_url}/")
```
Нужно исправить: здесь и далее: в тестах и методах пейджей не должно быть обращений к драйверу и использований WebdriverWait. Все подобные реализации должны находиться в рамках базового класса пейджей.

ShamilovR commented on pages/main_page.py:
```diff
@@ -0,0 +1,43 @@
+
+    def open(self):
+        self.driver.get(f"{self.base_url}/")
+        return self
```
Можно лучше: возвращать self из метода необязательно. Это может быть актуально при использовании паттерна chain of calls. Однако, он здесь не используется.

ShamilovR commented on tests/test_constructor.py:
```diff
@@ -0,0 +1,21 @@
+        main_page = MainPage(driver)
+        main_page.open()
+        main_page.go_to_order_feed()
+        assert "feed" in driver.current_url
```
Нужно исправить: здесь и далее: в тестах и методах пейджей не должно быть обращений к драйверу и использований WebdriverWait. Все подобные реализации должны находиться в рамках базового класса пейджей.

Gabaraeva committed:
Complete UI tests with all fixes

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,58 @@
+
+    main_page.open()
+    main_page.go_to_login()
+    login_page.login("test@example.com", "password")
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,58 @@
+
+    # Проверяем успешность логина
+    assert main_page.is_constructor_displayed(), "Login failed"
+    yield driver
```
Можно лучше: если в фикстуре после yield ничего не выполняется, то правильнее будет использовать return

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,58 @@
+    login_page.login("test@example.com", "password")
+
+    # Проверяем успешность логина
+    assert main_page.is_constructor_displayed(), "Login failed"
```
Можно лучше: проверки в фикстурах избыточны. Они должны производиться в рамках тестовых методов

ShamilovR commented on pages/main_page.py:
```diff
@@ -0,0 +1,77 @@
+
+    @allure.step("Открыть главную страницу")
+    def open(self):
+        self.driver.get(f"{self.base_url}/")
```
Нужно исправить: здесь и далее: в тестах и методах пейджей не должно быть обращений к драйверу и использований WebdriverWait. Все подобные реализации должны находиться в рамках базового класса пейджей.

Gabaraeva committed:
Refactor project:
- Remove direct driver access
- Use BasePage methods everywhere
- Encapsulate browser interactions

ShamilovR commented on pages/base_page.py:
```diff
@@ -0,0 +1,78 @@
+class BasePage:
+    def __init__(self, driver: WebDriver):
+        self.driver = driver
+        self.base_url = "https://stellarburgers.nomoreparties.site"
```
Нужно исправить: базовый класс это абстракция. Он не должен работать с конкретными элементами, локаторами и урлами.
